src/interfaces/wallet.cpp

```diff
@@ -135,55 +135,55 @@ WalletTxOut MakeWalletTxOut(interfaces::Chain::Lock& locked_chain,
 class WalletImpl : public Wallet
 {
 public:
-    explicit WalletImpl(const std::shared_ptr<CWallet>& wallet) : m_shared_wallet(wallet), m_wallet(*wallet.get()) {}
+    explicit WalletImpl(const std::shared_ptr<CWallet>& wallet) : m_wallet(wallet) {}
 
     bool encryptWallet(const SecureString& wallet_passphrase) override
     {
-        return m_wallet.EncryptWallet(wallet_passphrase);
+        return m_wallet->EncryptWallet(wallet_passphrase);
     }
-    bool isCrypted() override { return m_wallet.IsCrypted(); }
-    bool lock() override { return m_wallet.Lock(); }
-    bool unlock(const SecureString& wallet_passphrase) override { return m_wallet.Unlock(wallet_passphrase); }
-    bool isLocked() override { return m_wallet.IsLocked(); }
+    bool isCrypted() override { return m_wallet->IsCrypted(); }
+    bool lock() override { return m_wallet->Lock(); }
+    bool unlock(const SecureString& wallet_passphrase) override { return m_wallet->Unlock(wallet_passphrase); }
+    bool isLocked() override { return m_wallet->IsLocked(); }
     bool changeWalletPassphrase(const SecureString& old_wallet_passphrase,
         const SecureString& new_wallet_passphrase) override
     {
-        return m_wallet.ChangeWalletPassphrase(old_wallet_passphrase, new_wallet_passphrase);
+        return m_wallet->ChangeWalletPassphrase(old_wallet_passphrase, new_wallet_passphrase);
     }
-    void abortRescan() override { m_wallet.AbortRescan(); }
-    bool backupWallet(const std::string& filename) override { return m_wallet.BackupWallet(filename); }
-    std::string getWalletName() override { return m_wallet.GetName(); }
+    void abortRescan() override { m_wallet->AbortRescan(); }
+    bool backupWallet(const std::string& filename) override { return m_wallet->BackupWallet(filename); }
+    std::string getWalletName() override { return m_wallet->GetName(); }
     bool getKeyFromPool(bool internal, CPubKey& pub_key) override
     {
-        return m_wallet.GetKeyFromPool(pub_key, internal);
+        return m_wallet->GetKeyFromPool(pub_key, internal);
     }
-    bool getPubKey(const CKeyID& address, CPubKey& pub_key) override { return m_wallet.GetPubKey(address, pub_key); }
-    bool getPrivKey(const CKeyID& address, CKey& key) override { return m_wallet.GetKey(address, key); }
-    bool isSpendable(const CTxDestination& dest) override { return IsMine(m_wallet, dest) & ISMINE_SPENDABLE; }
-    bool haveWatchOnly() override { return m_wallet.HaveWatchOnly(); };
+    bool getPubKey(const CKeyID& address, CPubKey& pub_key) override { return m_wallet->GetPubKey(address, pub_key); }
+    bool getPrivKey(const CKeyID& address, CKey& key) override { return m_wallet->GetKey(address, key); }
+    bool isSpendable(const CTxDestination& dest) override { return IsMine(*m_wallet.get(), dest) & ISMINE_SPENDABLE; }
```

Comment:
Here and other places, this should change to `*m_wallet` instead of `*m_wallet.get()`. It's shorter, more readable, and potentially allows better error checking since the overloaded method can detect a null dereference.

Reply:
Oh I forgot about that operator, thanks.